netflix-roulette/src/data/HomePageDataProvider.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { Movie } from "@utils/interface";
+import { GenreType, MOVIES } from "@utils/constants";
+
+export type HomePageContextProps = {
+  movies: Movie[];
+  setMovies: (filteredMovies: Movie[]) => void;
+  selectedGenre: GenreType;
+  setSelectedGenre: (selectedGenre: GenreType) => void;
+  searchText: string;
+  setSearchText: (searchText: string) => void;
+  isAddMovieDialogVisible: boolean;
+  setIsAddMovieDialogVisible: (isMovieDetailsDialogVisible: boolean) => void;
+  movieToEdit: Movie;
+  setMovieToEdit: (movieToEdit: Movie) => void;
+  movieToDelete: Movie;
+  setMovieToDelete: (movieToEdit: Movie) => void;
+  movieToView: Movie;
+  setMovieToView: (movieToView: Movie) => void;
+};
+
+export const HomePageContext = React.createContext<HomePageContextProps>(
+  {} as any
+);
+
+export const HomePageDataProvider: React.FC = ({ children }) => {
+  const [movies, setMovies] = React.useState<Movie[]>([]);
+  const [searchText, setSearchText] = React.useState<string>("");
+  const [selectedGenre, setSelectedGenre] = React.useState<GenreType>(
+    GenreType.ALL
+  );
+  const [isAddMovieDialogVisible, setIsAddMovieDialogVisible] =
+    React.useState(false);
+  const [movieToEdit, setMovieToEdit] = React.useState<Movie>(null);
+  const [movieToDelete, setMovieToDelete] = React.useState<Movie>(null);
+  const [movieToView, setMovieToView] = React.useState<Movie>(null);
+
+  React.useEffect(() => setMovies(MOVIES), []);
+
+  const context = React.useMemo(
```

Comment:
If the purpose of this one was to avoid unnecessary re-rendering, I guess, it would make more sense to wrap with `useMemo()` _consumed_ parts of the context (inside individual components), so that when _certain_ parts of the context are changed it won't trigger re-render of components, depending on _untouched_ parts of the context. 

In this particular implementation, whenever _any_ of supplied dependency changes, it will trigger recalc of entire `context` which will lead to changed reference and all the dependencies re-rendered.